### PR TITLE
[Mobile Payments] Show IPP address errors when using site credentials

### LINKED ISSUE #14333: [Mobile Payments] Errors for empty Store Address/Postcode aren't shown when using site credentials
## Description

A user opened a ticket because they couldn't connect their card reader, they just got a generic error when they tried. There were no useful errors in the app or backend logs.

Underlying errors should have been logged, but we found that call was missing from the StripeCardReaderService `fetchDefaultLocationID` failure path. That should be fixed.

We determined that they were logged in using site credentials, and when trying to connect a reader on their site we found that there was indeed an error when calling `fetchDefaultLocationID`.

```
error	Networking.NetworkError	unacceptableStatusCode
Sorry, there was an issue with the server. Please try again later. (Error code: 500). Response: {"code":"store_address_is_incomplete","message":"https:\/\/{SITE_URL_REMOVED}\/wp-admin\/admin.php?page=wc-settings&tab=general","data":null}
```

This error is expected to be transformed to a `CardReaderConfigError.incompleteStoreAddress`, but that is gated by a check for the error type – it only works for `DotcomError`, not `NetworkError`, and because we're using site credentials, we'll always get a `NetworkError` here.

## Repro

1. Remove the Store Address or Postcode in WPAdmin settings.
2. Launch the app and log in to the site **using site credentials**, not Jetpack
3. Attempt to connect a card reader
4. Observe that you're shown a generic error

Expectation: you should see a specific error telling you to update the store address or postcode.

## Suggested remedies
1. Decode the expected error for both `NetworkError` and `DotcomError`.
2. Fix missing `UnderlyingError` logging in `StripeCardReaderService`.
3. Check for other `DotcomError` specific use and log tickets as needed.
4. Start team discussion about logging all API errors in the Network layer.

### PR DESCRIPTION
**Pull request by joshheald**
Closes: #14333

## Description

This PR fixes an issue found while helping a merchant through support, who couldn't connect to their readers with a new store when logged in using site credentials.

### Background
Previously, we decoded specific `DotcomError` instances to show the specific errors needed when a user hasn’t set their site address or postcode in WP-Admin, which prevents IPP reader connections from working.

These errors weren’t decoded when using site credentials to log in, as in that case a `NetworkError` is received, and our error handling didn’t decode that correctly.

This adds correct decoding and tests for the `NetworkError` variants of these issues.

## Steps to reproduce

1. Remove the Store Address or Postcode in WPAdmin settings.
2. Launch the app and log in to the site using site credentials, not Jetpack
3. Attempt to connect a card reader
4. Observe that you're shown a specific error with a button to update your address
5. Observe that specific error is logged in the console/application logs.

## Testing information

**Note** – I'm not able to get the postal code specific error to show; the backend doesn't appear to actually return this error, at least not for WCPay sites. It might still do it for Stripe sites, so I've covered the logic as it previously existed, and added tests, but not seen it with a real site.

Tested with an iPhone 15 Pro Max on iOS 18.0 using an M2 card reader.

I checked with both site credentials and Jetpack login.

### Logs
Console logs showing the underlying error and connection error

### Before

https://github.com/user-attachments/assets/1cae79dc-e95d-4a80-b638-3721e22e891e

### After

https://github.com/user-attachments/assets/e0613fbf-ece5-4c8c-ad9c-40af7486fc67

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/231">21.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Yosemite/YosemiteTests/Tools/CommonReaderConfigProviderTests.swift [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        await #expect(throws: expectedError) {
+            try await withCheckedThrowingContinuation { continuation in
+                sut.fetchDefaultLocationID { result in
+                    continuation.resume(with: result)
+                }
+            }
+        }
+    }
+
+    @Test func fetchDefaultLocationID_returns_invalidPostalCode_error_when_jetpack_site_has_no_postcode() async throws {
```
Comment (staskus) [inline Yosemite/YosemiteTests/Tools/CommonReaderConfigProviderTests.swift:28]: Nice. I haven't yet tried it!

**Review by staskus — APPROVED** [review]
👍 Excellent PR, thanks for the fix.

File: Yosemite/YosemiteTests/Tools/CommonReaderConfigProviderTests.swift (hunk 1 above)
Comment (joshheald) [reply Yosemite/YosemiteTests/Tools/CommonReaderConfigProviderTests.swift:28]: It was really easy to get started with, I can't see any reason that we shouldn't use it for new test files (and maybe just all new tests)

**Pull request merged**